### PR TITLE
Add `Leading by:` feature to Leaderboard

### PR DESCRIPTION
**Pull request by sassypocoyo**
Added the `Leading by:` feature to the Leaderboard, which enables different Leaderboards based on different variables. This allowed me to create a new Distance Leaderboard, so we can now see who has placed the farthest pin from the CeSIUM Headquarters!

Also made some improvements to some functions.

**Preview:**

image

### CODE REVIEW

**Review by RuiL1904 — COMMENTED** [review]
I think it would make more sense if the "leading by distance" was the sum of all the distances of the user. Otherwise, this feature it's the same as "sort locations by distance descending".

**Comment by sassypocoyo** [comment]
> I think it would make more sense if the "leading by distance" was the sum of all the distances of the user. Otherwise, this feature it's the same as "sort locations by distance descending".

It does make sense to sum the distances of all pins, it makes the "game" more interesting and takes into account the amount of pins that user has. Otherwise there would be a limit to how far a pin could be, and that author would forever be the Distance Leaderboard leader until someone traveled farther.

Anyway, implemented that and also added a Info icon to inform the user on how the distances are calculated and a hover text to the authors name saying "Go to <authors-name>'s GitHub".

File: components/Leaderboard/style.module.css [hunk 1]
```diff
@@ -68,28 +68,78 @@
...
+  user-select: none;
+  -webkit-user-select: none;
+  touch-action: manipulation;
+  width: fit-content;
+  word-break: break-word;
+  border: 0;
+  margin: 0 3px 0 3px;
+}
+
+.leading_info {
```
Comment (nelsonmestevao) [inline components/Leaderboard/style.module.css:141]: We should probably write our CSS module classes in PascalCase in the future :bulb:

Comment (sassypocoyo) [reply components/Leaderboard/style.module.css:141]: Noted, will make sure to do a PR in the future changing all CSS to PascalCase :muscle:

Comment (nelsonmestevao) [reply components/Leaderboard/style.module.css:141]: That would be awesome 🚀

**Review by RuiL1904 — APPROVED** [review]
Nice work! 👏🏼

**Pull request merged**